pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -0,0 +1,242 @@
+
+		var claimErrs []error
+		for _, obj := range objs {
+			objectMeta, err := meta.Accessor(obj)
+			if err != nil {
+				claimErrs = append(claimErrs, fmt.Errorf("error getting meta accessor for %T: %w", obj, err))
+				continue
+			}
+
+			patchLogger := claimLogger.WithValues("objectNamespace", objectMeta.GetNamespace(), "objectName", objectMeta.GetName())
+			patchLogger.V(4).Info("patching to get claim labels updated")
+
+			// Empty patch, allowing the admission plugin to update the resource to the correct labels
+			err = c.patchGenericObject(ctx, objectMeta, gvr, clusterName)
+			if err != nil {
+				gotPatchError = true
+				claimErrs = append(claimErrs, fmt.Errorf("error patching %q %s|%s/%s: %w", gvr, clusterName, objectMeta.GetNamespace(), objectMeta.GetName(), err))
+				continue
+			}
+		}
+
+		if len(claimErrs) > 0 {
+			allErrors = append(allErrors, claimErrs...)
+
+			if acceptedClaims.Has(s) {
+				applyErrors.Insert(s)
+			}
+		}
+	}
+
+	var unexpectedOrInvalidErrors []error
+	for _, s := range unexpectedClaims.List() {
+		claim, err := apisv1alpha1.PermissionClaimFrom(s)
+		if err != nil {
+			unexpectedOrInvalidErrors = append(unexpectedOrInvalidErrors, fmt.Errorf("error parsing claim string %q: %w", s, err))
+			continue
+		}
+		unexpectedOrInvalidErrors = append(unexpectedOrInvalidErrors, fmt.Errorf("unexpected/invalid claim for %s.%s (identity %q)", claim.Resource, claim.Group, claim.IdentityHash))
+	}
+	if len(unexpectedOrInvalidErrors) > 0 {
+		conditions.MarkFalse(
+			apiBinding,
+			apisv1alpha1.PermissionClaimsValid,
+			apisv1alpha1.InvalidPermissionClaimsReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Unexpected and/or invalid permission claims: %s",
+			aggregateerrors.NewAggregate(unexpectedOrInvalidErrors),
+		)
+	} else {
+		conditions.MarkTrue(apiBinding, apisv1alpha1.PermissionClaimsValid)
+	}
+
+	fullyApplied := expectedClaims.Difference(applyErrors)
+	apiBinding.Status.AppliedPermissionClaims = []apisv1alpha1.PermissionClaim{}
+	for _, s := range fullyApplied.List() {
+		claim, err := apisv1alpha1.PermissionClaimFrom(s)
+		if err != nil {
+			allErrors = append(allErrors, fmt.Errorf("error parsing fully applied claim %q: %w", s, err))
+			continue
+		}
+
+		apiBinding.Status.AppliedPermissionClaims = append(apiBinding.Status.AppliedPermissionClaims, *claim)
+	}
+
+	if len(allErrors) > 0 {
+		msg := aggregateerrors.NewAggregate(allErrors).Error()
+		if len(msg) > 100 {
+			msg = msg[0:99] + "..."
+		}
+
+		conditions.MarkFalse(
+			apiBinding,
+			apisv1alpha1.PermissionClaimsApplied,
+			apisv1alpha1.InternalErrorReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Permission claims have not been fully applied: %s",
+			msg,
+		)
+	} else {
+		conditions.MarkTrue(apiBinding, apisv1alpha1.PermissionClaimsApplied)
+	}
+
+	if gotPatchError {
+		return fmt.Errorf("got at least 1 patch error for %s|%s: %v", clusterName, apiBinding.Name, allErrors)
+	}
```

Comment:
here we should use a sample, e.g. first 10 patch errors, not all 3829

Reply:
will do